app/com/arpnetworking/metrics/portal/alerts/impl/CachingAlertExecutionRepository.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2021 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import akka.actor.ActorRef;
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.commons.java.time.TimeAdapters;
+import com.arpnetworking.commons.java.util.concurrent.CompletableFutures;
+import com.arpnetworking.metrics.portal.alerts.AlertExecutionRepository;
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import models.internal.Organization;
+import models.internal.alerts.AlertEvaluationResult;
+import models.internal.scheduling.JobExecution;
+import net.sf.oval.constraint.NotNull;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.inject.Named;
+
+/**
+ * An alert repository wrapper that caches the most recent successful results.
+ * <br>
+ * {@link AlertExecutionRepository#getLastSuccessBatch} is probably the most frequently
+ * accessed piece of execution data for alerts, so it's useful to cache those results
+ * separately.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public final class CachingAlertExecutionRepository implements AlertExecutionRepository {
+    private final AlertExecutionRepository _inner;
+    private final ActorRef _successCache;
+    private final Duration _cacheTimeout;
```

Comment:
This variable is a bit misleading; it's the cache ttl, it's the cache operation timeout.